tests/kafkatest/tests/connect/connect_rest_test.py

```diff
@@ -65,7 +66,9 @@ def test_rest_api(self):
         sink_connector_props = self.render("connect-file-sink.properties")
         for connector_props in [source_connector_props, sink_connector_props]:
             connector_config = self._config_dict_from_props(connector_props)
-            self.cc.create_connector(connector_config)
+            wait_until(lambda: self.not_throw_rest_error(lambda: self.cc.create_connector(connector_config)), timeout_sec=120, err_msg="Create connectors throws exception.")
```

Comment:
I think we might need to address the problem more generally. Technically you can get 409 errors on pretty much any call due to rebalances, right? Or did you choose to protect the specific API calls you did for a reason? Are there only certain cases that can trigger rebalances in this test given that we aren't bouncing any workers and therefore only protecting a subset of them is safe?

I'm wondering if the approach to retrying should be flipped around, where every API call has something like a `retries` parameter. That would allow for easily indicating you expect immediate success (retries=0) or that you want to account for possible rebalances (retries=some larger number).